sunpy/map/mapbase.py

```diff
@@ -518,6 +518,29 @@ def _new_instance(cls, data, meta, plot_settings=None, **kwargs):
             new_map.plot_settings.update(plot_settings)
         return new_map
 
+    @classmethod
+    def notes_fix( cls,notes_doc, datadoc):
+        existing_notes_pos = datadoc.find('Notes\n    -----')
+        existing_notes_pos2 = notes_doc.find('Notes\n-----')
+        existing_notes_data = textwrap.indent(notes_doc[existing_notes_pos2 + len('Notes\n-----'):].strip(), "    ")
+        references_pattern = "References\n    ----------"
+        examples_pattern = "Examples\n   -------"
+        start_index = existing_notes_pos + len('Notes\n    -----')
+        references_pos = datadoc.find(references_pattern, start_index)
+        examples_pos = datadoc.find(examples_pattern, start_index)
+        if references_pos != -1 or examples_pos != -1:
+            next_pattern_pos = min(pos for pos in [references_pos, examples_pos] if pos != -1)
+            other_patterns = datadoc[:next_pattern_pos]
+            if existing_notes_pos!=-1:
+                datadoc = other_patterns + existing_notes_data.lstrip() + '\n    ' + datadoc[next_pattern_pos:]
+            else:
+                datadoc = other_patterns + 'Notes\n    -----\n' + existing_notes_data + '\n    ' + datadoc[next_pattern_pos:]
+        elif existing_notes_pos != -1 and (references_pos == -1 or examples_pos == -1):
+            datadoc +="\n"+existing_notes_data
+        else:
+            datadoc += textwrap.indent(notes_doc, "    ")
+        return datadoc
```

Comment:
```suggestion
    def fix_duplicate_notes(cls, notes, data):
    existing_notes_pos_data = notes.find('Notes\n-----')
    existing_notes_data = textwrap.indent(notes[existing_notes_pos_data + len('Notes\n-----'):].strip(), "    ")
    references_pattern = "References\n    ----------"
    examples_pattern = "Examples\n   -------"
    start_index = doc.find(references_pattern if references_pattern in doc else examples_pattern)
    
    if start_index != -1:
        next_pattern_pos = min(pos for pos in [doc.find(references_pattern, start_index), doc.find(examples_pattern, start_index)] if pos != -1)
        other_patterns = doc[:next_pattern_pos]
        
        if existing_notes_pos_data != -1:
            doc = other_patterns + existing_notes_data.lstrip() + '\n    ' + doc[next_pattern_pos:]
        else:
            doc = other_patterns + 'Notes\n    -----\n' + existing_notes_data + '\n    ' + doc[next_pattern_pos:]
    elif existing_notes_pos_data != -1:
        doc += "\n" + existing_notes_data
    else:
        doc += textwrap.indent(notes, "    ")
    
    return doc
```

Can you see if this works?

Reply:
yea this also works!